thingif/src/main/java/com/kii/thingif/query/AggregatedResultAdapter.java

```diff
@@ -0,0 +1,59 @@
+package com.kii.thingif.query;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.kii.thingif.TargetState;
+import com.kii.thingif.internal.gson.HistoryStateAdapter;
+import com.kii.thingif.internal.gson.TimeRangeAdapter;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AggregatedResultAdapter<T extends Number, S extends TargetState>
+        implements JsonDeserializer<AggregatedResult<T, S>> {
+    private Class<S> stateClass;
+
+    public AggregatedResultAdapter(Class<S> cls) {
+        this.stateClass = cls;
+    }
+
+    @Override
+    public AggregatedResult<T, S> deserialize(
+            JsonElement jsonElement,
+            Type typeOfT,
+            JsonDeserializationContext context)
+        throws JsonParseException
+    {
+        if (jsonElement == null) return null;
+
+        JsonObject json = jsonElement.getAsJsonObject();
+
+        if (!json.has("range") || !json.has("value")) {
+            throw new JsonParseException("no neceessary fields.");

```

Comment:
change `no neceessary fields.` to `json doesn't contain both of 2 necessary fields: range and value. `